pypeit/images/combineimage.py

```diff
@@ -144,53 +258,102 @@ def run(self, bias=None, flatimages=None, ignore_saturation=False, sigma_clip=Tr
+            # First calculate the error in the sum.  The variance is set to 0
+            # for pixels masked in all images.
+            comb_rn2 = np.ma.sum(np.ma.MaskedArray(rn2img_stack, mask=bpm_stack),axis=0).filled(0.)
+            comb_procv = np.ma.sum(np.ma.MaskedArray(procv_stack, mask=bpm_stack),axis=0).filled(0.)
+            # Convert to standard error in the median (pi/2 factor relates standard variance
+            # in mean (sum(variance_i)/n^2) to standard variance in median)
+            comb_rn2[gpm] *= np.pi/2/nstack[gpm]**2/comb_scl[gpm]**2
+            comb_procv[gpm] *= np.pi/2/nstack[gpm]**2/comb_scl[gpm]**2
         else:
+            # NOTE: Given the check at the beginning of the function, the code
+            # should *never* make it here.
             msgs.error("Bad choice for combine.  Allowed options are 'median', 'weightmean'.")
 
-        # Build the last one
-        final_pypeitImage = pypeitimage.PypeItImage(img_list_out[0],
-                                                    ivar=utils.inverse(var_list_out[0]),
-                                                    bpm=pypeitImage.bpm,
-                                                    rn2img=var_list_out[1],
-                                                    crmask=np.logical_not(gpm),
-                                                    detector=pypeitImage.detector,
-                                                    PYP_SPEC=pypeitImage.PYP_SPEC)
+        # Recompute the inverse variance using the combined image
+        comb_var = procimg.variance_model(comb_rn2, counts=comb_img, darkcurr=comb_dark,
+                                          exptime=comb_texp, proc_var=comb_procv,
+                                          count_scale=comb_scl,
+                                          noise_floor=self.par['noise_floor'],
+                                          shot_noise=self.par['shot_noise'])
+
+        # Build the combined image
+        comb = pypeitimage.PypeItImage(image=comb_img, ivar=utils.inverse(comb_var), nimg=nstack,
+                                       rn2img=comb_rn2, proc_var=comb_procv, img_scale=comb_scl,
+                                       # NOTE: The detector is needed here so
+                                       # that we can get the dark current later.
+                                       bpm=bpm, detector=pypeitImage.detector,
+                                       PYP_SPEC=self.spectrograph.name,
+                                       units='e-' if self.par['apply_gain'] else 'ADU',
+                                       exptime=comb_texp, noise_floor=self.par['noise_floor'],
+                                       shot_noise=self.par['shot_noise'])
+
         # Internals
-        final_pypeitImage.rawheadlist = pypeitImage.rawheadlist
-        final_pypeitImage.process_steps = pypeitImage.process_steps
+        # TODO: Do we need these two?
+        comb.rawheadlist = pypeitImage.rawheadlist
```

Comment:
I bet we need the `rawheadlist`

Reply:
Okay.  I'll leave them both for now, but also leave the comment.